qa/integration-tests/src/test/java/io/camunda/it/client/DecisionInstanceQueryTest.java

```diff
@@ -0,0 +1,274 @@
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.it.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import io.camunda.it.utils.BrokerWithCamundaExporterITInvocationProvider;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.command.ProblemException;
+import io.camunda.zeebe.client.api.response.DeploymentEvent;
+import io.camunda.zeebe.client.api.response.EvaluateDecisionResponse;
+import io.camunda.zeebe.client.api.search.response.DecisionDefinitionType;
+import io.camunda.zeebe.client.api.search.response.DecisionInstance;
+import io.camunda.zeebe.client.api.search.response.DecisionInstanceState;
+import io.camunda.zeebe.client.protocol.rest.BasicLongFilterProperty;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@TestInstance(Lifecycle.PER_CLASS)
+@ExtendWith(BrokerWithCamundaExporterITInvocationProvider.class)
+class DecisionInstanceQueryTest {
+
+  private static final String DECISION_DEFINITION_ID_1 = "decision_1";
+  private static final String DECISION_DEFINITION_ID_2 = "invoiceAssignApprover";
+  // evaluated decisions mapped by decision definition id
+  private final Map<String, EvaluateDecisionResponse> evaluatedDecisions = new HashMap<>();
+  private boolean initialized;
+
+  @BeforeEach
+  void setUp(final ZeebeClient zeebeClient) {
+    if (!initialized) {
+
+      List.of("decision_model.dmn", "invoiceBusinessDecisions_v_1.dmn")
+          .forEach(
+              dmn ->
+                  deployResource(zeebeClient, String.format("decisions/%s", dmn)).getDecisions());
+      evaluatedDecisions.put(
+          DECISION_DEFINITION_ID_1,
+          evaluateDecision(
+              zeebeClient, DECISION_DEFINITION_ID_1, "{\"age\": 20, \"income\": 20000}"));
+      evaluatedDecisions.put(
+          DECISION_DEFINITION_ID_2,
+          evaluateDecision(
+              zeebeClient,
+              DECISION_DEFINITION_ID_2,
+              "{\"amount\": 100, \"invoiceCategory\": \"Misc\"}"));
+      waitForDecisionsToBeEvaluated(
+          zeebeClient,
+          evaluatedDecisions.values().stream()
+              .mapToInt(v -> v.getEvaluatedDecisions().size())
+              .sum());
+      initialized = true;
+    }
+  }
```

Comment:
❓ Is there anything that prevents us from doing a `BeforeAll` which is executed once for the test class? We are now effectively running this once for the test class, right?

Reply:
I tried to use @BeforeAll but could not have the parameter resolver working 
`org.junit.jupiter.api.extension.ParameterResolutionException: No ParameterResolver registered for parameter [final io.camunda.zeebe.client.ZeebeClient zeebeClient] in method ...`
It seems that [this](https://github.com/camunda/camunda/blob/1753c1efd1a5480847d411f19c77f5536e06cf96/qa/integration-tests/src/test/java/io/camunda/it/utils/BrokerWithCamundaExporterITInvocationProvider.java#L147) is invoked afterward.
Probably, `BeforeAll` is not compatible with ParameterResolver used within the context of `@TestTemplate`.

I haven’t examined this in detail, but I prefer not to spend too much time on it.